tests/core/policies/test_intent_ted_policy.py

```diff
@@ -229,58 +369,58 @@ def test_unlikely_intent_check(
         # Record the original thresholds because the test will modify it.
         # Need to revert it back so that the `trained_policy` can
         # be reused across tests.
-        original_label_thresholds = trained_policy.label_thresholds
+        original_label_thresholds = loaded_policy.label_thresholds
 
-        trained_policy.label_thresholds[dummy_intent_index] = threshold_value
+        loaded_policy.label_thresholds[dummy_intent_index] = threshold_value
         query_intent = default_domain.intents[dummy_intent_index]
 
-        unlikely_intent_prediction = trained_policy._check_unlikely_intent(
+        unlikely_intent_prediction = loaded_policy._check_unlikely_intent(
             default_domain, similarities, query_intent
         )
 
         assert is_unlikely == unlikely_intent_prediction
 
         # Revert back the label thresholds to original values.
-        trained_policy.label_thresholds = original_label_thresholds
+        loaded_policy.label_thresholds = original_label_thresholds
```

Comment:
Is modifying a fixture safe to do? Tests can be run with multiple threads or tests could fail before resetting? Perhaps we should clone the fixture at the start of the test instead?

Reply:
Removed the fixture. Instead using a common function which persists and loads the `trained_policy`(which is a fixture). All changes are made to the loaded policy and hence the fixture remains untouched.